src/interceptors/ClientRequest/NodeClientRequest.test.ts

```diff
@@ -315,3 +315,26 @@ it('does not send request body to the original server given mocked response', as
   const text = await getIncomingMessageBody(response)
   expect(text).toBe('mock created!')
 })
+
+it.only('emits the ERR_STREAM_WRITE_AFTER_END error when write after end given no mocked response', async () => {
```

Comment:
I think this should be a compliance test in `test/modules/http/compliance` where we will also feature the real usage of `http.*` instead of constructing the internal class.